src/interfaces/assistants_web/src/components/MessageRow/MessageRow.tsx

```diff
@@ -56,7 +57,13 @@ export const MessageRow = forwardRef<HTMLDivElement, Props>(function MessageRowI
 
   const [isShowing, setIsShowing] = useState(false);
   const [isLongPressMenuOpen, setIsLongPressMenuOpen] = useState(false);
+  const { showSteps } = useSettingsStore();
   const [isStepsExpanded, setIsStepsExpanded] = useState<boolean>(true);
+
+  useEffect(() => {
+    setIsStepsExpanded(showSteps);
+  }, [showSteps]);
+
```

Comment:
```suggestion
  const [isStepsExpanded, setIsStepsExpanded] = useState<boolean>(showSteps);
```

Reply:
@abimacarmes I tried this change originally, but was dealing with a problem where `isStepsExpanded` wasn't being updated subsequently, but the `useEffect` seemed to do the trick